perf/ring.go

```diff
@@ -86,17 +86,25 @@ func (ring *perfEventRing) Close() {
 	ring.mmap = nil
 }
 
-func createPerfEvent(cpu, watermark int) (int, error) {
-	if watermark == 0 {
-		watermark = 1
+func createPerfEvent(cpu, wakeupBytes int, wakeupEvents int) (int, error) {
```

Comment:
This should return an error if both bytes and events are set.

Reply:
This should probably take a `ReaderOptions` instead of more ints.